test/integ/test_multi_statement.py

```diff
@@ -0,0 +1,385 @@
+            q_id = cur.execute_async(sql).get("queryId")
+            with pytest.raises(ProgrammingError) as sync_error:
+                cur.execute(sql)
+            while con.is_still_running(con.get_query_status(q_id)):
+                time.sleep(1)
+            assert con.get_query_status(q_id) == QueryStatus.FAILED_WITH_ERROR
+            with pytest.raises(ProgrammingError) as e1:
+                con.get_query_status_throw_if_error(q_id)
+            assert sync_error.value.errno != -1
+            with pytest.raises(ProgrammingError) as e2:
+                cur.get_results_from_sfqid(q_id)
+            assert e1.value.errno == e2.value.errno == sync_error.value.errno
+
+
+def test_mix_sync_async_multi(conn_cnx):
+    """Tests sending multiple multi-statement async queries at the same time."""
+    with conn_cnx(
+        session_parameters={
+            PARAMETER_MULTI_STATEMENT_COUNT: 0,
+        }
+    ) as con:
+        with con.cursor() as cur:
+            cur.execute("alter session set CLIENT_TIMESTAMP_TYPE_MAPPING=TIMESTAMP_TZ")
+            cur.execute(
+                "create or replace temp table smallTable (colA string, colB int);"
+                "create or replace temp table uselessTable (colA string, colB int);"
+            )
+            for table in ["smallTable", "uselessTable"]:
+                cur.execute(
+                    "insert into {t} values('row1', 1);"
+                    "insert into {t} values('row2', 2);"
+                    "insert into {t} values('row3', 3);".format(t=table)
+                )
+            cur.execute_async("select 1; select 'a'; select * from smallTable;")
+            sf_qid1 = cur.sfqid
+            cur.execute_async("select 2; select 'b'; select * from uselessTable")
+            sf_qid2 = cur.sfqid
+            # Wait until the 2 queries finish
+            while con.is_still_running(con.get_query_status(sf_qid1)):
+                time.sleep(1)
+            while con.is_still_running(con.get_query_status(sf_qid2)):
+                time.sleep(1)
+            cur.execute("drop table uselessTable")
+            assert cur.fetchall() == [("USELESSTABLE successfully dropped.",)]
+            cur.get_results_from_sfqid(sf_qid1)
+            assert cur.fetchall() == [(1,)]
+            assert cur.nextset() == cur
+            assert cur.fetchall() == [("a",)]
+            assert cur.nextset() == cur
+            assert cur.fetchall() == [("row1", 1), ("row2", 2), ("row3", 3)]
+            assert cur.nextset() is None
+            assert not cur.fetchall()
+            cur.get_results_from_sfqid(sf_qid2)
+            assert cur.fetchall() == [(2,)]
+            assert cur.nextset() == cur
+            assert cur.fetchall() == [("b",)]
+            assert cur.nextset() == cur
+            assert cur.fetchall() == [("row1", 1), ("row2", 2), ("row3", 3)]
+            assert cur.nextset() is None
+            assert not cur.fetchall()
+
+
+def test_done_caching_multi(conn_cnx):
```

Comment:
I challenge you on this test. I think this is the same as https://github.com/snowflakedb/snowflake-connector-python/blob/88843f88a5b7de84bbca3587da86c702155fd96c/test/integ/test_async.py#L166 but you send in multi-statements which I think is not testing anything new.

Reply:
It's quite similar, I used it as inspiration because I wanted to be sure the retrieval of the individual query IDs in the multi-statement wasn't messing with the async QID caching in the connector. However, I realize I'm not even pulling in the multi-statement query IDs here since it's execute_async, so I'll add that.